### PR TITLE
Cppcheck github action, and style and bugfixes 

### PR DESCRIPTION
**Pull request by HybridDog**
I've used Cppcheck to find and fix some bugs and bad code style, and then added a github action to run the checks automatically. I haven't tested if the github action works.

The bugs:
* `AddonPreviewMenu::rebuild_menu` accessed the addon description string out of bounds.
* `TextObject::wrap_text` stripped all newlines, even double ones, because `prev_c` was not changed.
* `RainParticleSystem::set_amount` used `amount` instead of `real_amount` in the minimum calculation, which was probably not intended. I don't know how this affects the rain. 

Related issue: #1552
In the future when code is more clean, additional linters could be added, e.g. [TscanCode](https://github.com/Tencent/TscanCode) and Clang-Tidy. SuperTux apparently already has a `.clang-tidy` configuration file but no github action workflow for it.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#  SuperTux
+#  Copyright (C) 2022
```
Comment (mrkubax10) [inline .github/workflows/lint.yml:2]: GPL3 header is missing copyright holder

**Comment by HybridDog** [comment]
I think the new workflow works now and the 5 failing checks are not related to this PR and fail on other PRs, too.

**Review by mrkubax10 — APPROVED** [review]

**Comment by HybridDog** [comment]
Thank you for reviewing this. Can I merge it now?

**Comment by mrkubax10** [comment]
I'm not sure if I should decide about this but I think that you can

**Comment by HybridDog** [comment]
[CONTRIBUTING.md](https://github.com/SuperTux/supertux/blob/master/CONTRIBUTING.md) and the wiki don't mention conditions for merging (at least I cannot see them), so I think it should be fine.

**Pull request merged**